include/contractor/graph_contractor_adaptors.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef OSRM_CONTRACTOR_GRAPH_CONTRACTION_ADAPTORS_HPP_
+#define OSRM_CONTRACTOR_GRAPH_CONTRACTION_ADAPTORS_HPP_
+
+#include "contractor/contractor_graph.hpp"
+#include "util/log.hpp"
+
+#include <vector>
+
+namespace osrm
+{
+namespace contractor
+{
+
+// Make sure to move in the input edge list!
+template <typename InputEdgeContainer>
+std::vector<ContractorEdge> ContractorEdgeInputAdaptor(InputEdgeContainer input_edge_list)
+{
+    std::vector<ContractorEdge> edges;
+    edges.reserve(input_edge_list.size() * 2);
+
+    for (const auto &input_edge : input_edge_list)
+    {
+#ifndef NDEBUG
+        if (static_cast<unsigned int>(std::max(input_edge.weight, 1)) > 24 * 60 * 60 * 10)
```

Comment:
What kind of check would you have in mind? The entire edge-weight check is against a more or less random value to me. It's edges that take more than a day in time, but its not technically an overflow. The only thing that I could see is improving the logic to give a meaning to `24*60*60*10` in form of a `constexpr`.